src/main/java/com/integration/githubintegration/exception/GeneralExceptionHandler.java

```diff
@@ -0,0 +1,22 @@
+package com.integration.githubintegration.exception;
+
+import feign.FeignException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GeneralExceptionHandler {
+
+    @ExceptionHandler(FeignException.class)
+    public ResponseEntity<String> repositoryExceptionResponse(FeignException e) {
+        if (e.status() == 404) {
+            return ResponseEntity.status(404).body("Repository not found");
```

Comment:
Aktualnie fajne aczkolwiek co w sytuacji jesli mielibysmy wiecej niz jednego klienta? Czy na pewno kazdego chcielibysmy traktowac w ten sam sposob i dla kazdego sprawdzac z ktorego klienta poleciala 404 itp? 
Np w przypadku querowania Branchy tez by rzucilo napis Repository not found.
Dodatkowo jesli mielibysmy wiecej clientow do roznych serwisow to kazdy z nich rzucalby FeignException.
Zastanow sie czy nie lepszy rozwiazaniem byloby:
return ResponseEntity.status(404).body(e.getMessage());